Implemented GUI basics

ToxicCicity opened the pull request:
resolved #15 resolves #16 resolves #20 resolves #30

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

ToxicCicity committed:
removed unused import

ToxicCicity committed:
working on the first implementation of our gui. Experimental

azizcan1212 committed:
Merge branch 'toxicCicity' into infinity

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
fully implemented login for level 1 of our project

ToxicCicity committed:
Changes by pearow

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
added dynamic titling of the application according to the current user

ToxicCicity committed:
added new singleton pattern functionality

ToxicCicity committed:
finalized the base GUI implementation (no api requests included yet)

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

ToxicCicity committed:
1)started implementing api calls (api fixes not included in this commit). This means that some of the fake information crated and their respective methods are commented out
2)Added search functionality authored by @infinity1222
3)After login,GuiDeneme1 is now centered on screen
4)A default course and section are selected on initialization
5)Added the basis of Rate Professor functionality

ToxicCicity committed:
some code cleanup

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

# Conflicts:
#	src/main/resources/users.json

ToxicCicity committed:
last commit before full API utilization

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
Still not fully using API yet as the Review model and its respective functions are yet to be implemented. This commit includes some api calls and some code cleanup

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

ToxicCicity committed:
authored by @azizcan1212(infinity1222)
changed addRating's return type
added methods to save ratings of professors

ToxicCicity committed:
Merge branch 'refs/heads/master' into toxicCicity

ToxicCicity committed:
changed Course.addRating's return type to int

ToxicCicity committed:
Merge branch 'refs/heads/master' into toxicCicity

# Conflicts:
#	src/main/java/com/sen3006/coursewise/models/Course.java

Pearow committed:
Complete rating model

ToxicCicity committed:
Added and fixed review functionality to gui using api

ToxicCicity committed:
Changes by @Pearow, @azizcan1212 and commiter. Added and fixed api support for reviews

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

# Conflicts:
#	src/main/java/com/sen3006/coursewise/models/Professor.java

Pearow committed:
Add LecturersNote

ToxicCicity committed:
Changes by @Pearow (store currentCourse instead of currentCourseCode to decrease api calls)

ToxicCicity committed:
Added professor rating functionality using api

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

ToxicCicity committed:
Added first implementation of course note

ToxicCicity committed:
Added some TODO's

Pearow commented on src/main/java/com/sen3006/coursewise/GuiDeneme1.java:
```diff
@@ -0,0 +1,518 @@
+    private ToggleGroup sectionGroup;
+    private Course currentCourse;
+    private static boolean creatingNewReview = false;
+    private String currentSectionCode;
```
Store objects instead of ids

Pearow commented on src/main/java/com/sen3006/coursewise/GuiDeneme1.java:
```diff
@@ -0,0 +1,518 @@
+    }
+
+    //Add a section to the sections list
+    private void addSectionToList(String sectionCode) {
```
Don't add section codes to the list. Instead add sections so you can access data efficiently

Pearow commented on src/main/java/com/sen3006/coursewise/GuiDeneme1.java:
```diff
@@ -0,0 +1,518 @@
+        this.currentSectionCode = sectionCode;
+        courseCodeLabel.setText(currentCourse.getCourse_id());
+
+        for (Section s : api.getSections(currentCourse.getCourse_id())) {
```
Don't use for loops to access data. Getting section as a parameter is much much more efficient

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

Pearow committed:
Add CourseType

ToxicCicity committed:
optimized gui controller (changes by @azizcan1212)

ToxicCicity committed:
optimized gui controller

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

Pearow committed:
Fix duplicate hashing

ToxicCicity committed:
added section type support

ToxicCicity committed:
Merge branch 'refs/heads/pearow' into toxicCicity

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

azizcan1212 committed:
Merge branch 'refs/heads/toxicCicity' into infinity

azizcan1212 committed:
#32
Design and color change and term selection design(not functional) added

ToxicCicity committed:
1)Changed GuiDeneme1.fxml and GuiDeneme1.java to GUI and GUIController.java
2)Added better coloring
3)First steps for including term selection are commited

ToxicCicity committed:
new data files added
